diff --git a/src/gen/gen_vimdoc.lua b/src/gen/gen_vimdoc.lua
index 68912cf0b5..f3990cd9a6 100755
--- a/src/gen/gen_vimdoc.lua
+++ b/src/gen/gen_vimdoc.lua
@@ -1003,8 +1003,7 @@ local function gen_target(cfg)
       end
     end
     -- FIXME: Using f_base will confuse `_meta/protocol.lua` with `protocol.lua`
-    local f_base = vim.fs.basename(f)
-    sections[f_base] = make_section(f_base, cfg, briefs_txt, funs_txt)
+    sections[f] = make_section(f, cfg, briefs_txt, funs_txt)
   end
 
   local first_section_tag = sections[cfg.section_order[1]].help_tag
